src/Database/Table/ReferenceCache.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * This file is part of the Nette Framework (https://nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+declare(strict_types=1);
+
+namespace Nette\Database\Table;
+
+
+class ReferenceCache
+{
+	/** @var array */
+	protected $referencingPrototype = [];
+
+	/** @var array */
+	protected $referencing = [];
+
+	/** @var array */
+	protected $referenced = [];
+
+
+	public function &getReferencingPrototype(string $specificCacheKey, string $table, string $column)
+	{
+		return $this->referencingPrototype[$specificCacheKey]["$table.$column"];
+	}
+
+
+	public function clearReferencingPrototype(): void
+	{
+		$this->referencingPrototype = [];
+	}
+
+
+	/**
+	 * @return mixed
```

Comment:
It seems according to the code that it is `array|null`, or not? For `getReferenced` too.

Reply:
Yes, you are right